CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc

```diff
@@ -902,6 +903,37 @@ static function bool IsAcademyProjectActive(StateObjectReference FacilityRef)
 	return false;
 }
 
+static function string GetAcademyQueueMessage(StateObjectReference FacilityRef)
+{
+	local XComGameState_HeadquartersProjectTrainAcademy AcademyProject;
+	local XComGameState_FacilityXCom FacilityState;
+	local XComGameState_StaffSlot StaffSlot;
+	local string strStatus, Message;
+	local int i;
+
+	FacilityState = XComGameState_FacilityXCom(`XCOMHISTORY.GetGameStateForObjectID(FacilityRef.ObjectID));
+
+	for (i = 0; i < FacilityState.StaffSlots.Length; i++)
+	{
+		StaffSlot = FacilityState.GetStaffSlot(i);
+		if (StaffSlot.IsSlotFilled())
+		{
+			AcademyProject = class'X2Helper_Infiltration'.static.GetAcademyProjectForUnit(StaffSlot.GetAssignedStaffRef());
+			if (AcademyProject != none)
+			{
+				if (AcademyProject.GetCurrentNumHoursRemaining() < 0)
+					Message = class'UIUtilities_Text'.static.GetColoredText(class'UIFacility_Powercore'.default.m_strStalledResearch, eUIState_Warning);
+				else
+					Message = class'UIUtilities_Text'.static.GetTimeRemainingString(AcademyProject.GetCurrentNumHoursRemaining());
+
+				strStatus = StaffSlot.GetBonusDisplayString() $ ":" @ Message;
+				break;
```

Comment:
What happens if there are 2 people training currently?

Reply:
If it helps, I recently had 2 people training and still see the icon in the lower left about idle staff.